test/test_torch.py

```diff
@@ -3757,6 +3757,91 @@ def test_tril(self):
         torch.tril(x, out=res2)
         self.assertEqual(res1, res2, 0)
 
+    def _compare_trilu_indices(self, row, col, offset=0, dtype=torch.uint8):
+        if row == 0 or col == 0:
+            # have to handle this separately as tril and triu does not take
+            # empty matrix as input
+            self.assertEqual(
+                torch.empty(0, 2, dtype=dtype).transpose(0, 1),
+                torch.tril_indices(row, col, offset, dtype=dtype))
+
+            self.assertEqual(
+                torch.empty(0, 2, dtype=dtype).transpose(0, 1),
+                torch.triu_indices(row, col, offset, dtype=dtype))
+
+        else:
+            self.assertEqual(
+                torch.ones(row, col, dtype=dtype)
+                     .tril(offset).nonzero().transpose(0, 1),
+                torch.tril_indices(row, col, offset, dtype=dtype))
+
+            self.assertEqual(
+                torch.ones(row, col, dtype=dtype)
+                     .triu(offset).nonzero().transpose(0, 1),
+                torch.triu_indices(row, col, offset, dtype=dtype))
+
+    def test_tril_and_triu_indices(self):
+
```

Comment:
There's not actually a test for the default long behavior (calling tril_indices without a dtype argument), please add one or modify _compare_trilu_indices to not send an argument to torch.triu_indices.

Reply:
Will tests on line 3825-3833 be sufficient to cover that?

Reply:
Good point, that is sufficient. My bad I was reading that too fast